File: src/com/upokecenter/util/IRadixMathHelper.java
package com.upokecenter.util;

// import java.math.*;

  internal interface IRadixMathHelper<T> {
    int GetRadix();

    int GetArithmeticSupport();

    int GetSign(T value);

    int GetFlags(T value);

    BigInteger GetMantissa(T value);

    BigInteger GetExponent(T value);

    BigInteger RescaleByExponentDiff(BigInteger value, BigInteger exp1, BigInteger exp2);

    T ValueOf(int val);

    T CreateNewWithFlags(BigInteger mantissa, BigInteger exponent, int flags);

    IShiftAccumulator CreateShiftAccumulatorWithDigits(BigInteger value, int lastDigit, int olderDigits);

    IShiftAccumulator CreateShiftAccumulator(BigInteger value);

    boolean HasTerminatingRadixExpansion(BigInteger num, BigInteger den);

    BigInteger MultiplyByRadixPower(BigInteger value, FastInteger power);
  }

File: PeterO/Numbers/IRadixMathHelper.cs
/*
Written in 2014 by Peter O.
Any copyright is dedicated to the Public Domain.
http://creativecommons.org/publicdomain/zero/1.0/
If you like this, you should donate to Peter O.
at: http://upokecenter.dreamhosters.com/articles/donate-now-2/
 */
using System;

namespace PeterO.Numbers {
  internal interface IRadixMathHelper<T> {
    int GetRadix();

    int GetArithmeticSupport();

    int GetSign(T value);

    int GetFlags(T value);

    EInteger GetMantissa(T value);

    EInteger GetExponent(T value);

    T ValueOf(int val);

    T CreateNewWithFlags(EInteger mantissa, EInteger exponent, int flags);

    IShiftAccumulator CreateShiftAccumulatorWithDigits(
EInteger value,
int lastDigit,
int olderDigits);

    IShiftAccumulator CreateShiftAccumulator(EInteger value);

    bool HasTerminatingRadixExpansion(EInteger num, EInteger den);

    EInteger MultiplyByRadixPower(EInteger value, FastInteger power);
  }
}
